tests/functional/aws-node-sdk/test/multipleBackend/put.js

```diff
@@ -0,0 +1,181 @@
+            return bucketUtil.empty(bucket)
+            .then(() => {
+                process.stdout.write('Deleting bucket\n');
+                return bucketUtil.deleteOne(bucket);
+            })
+            .catch(err => {
+                process.stdout.write(`Error in afterEach: ${err}\n`);
+                throw err;
+            });
+        });
+
+        it('should put an object mem with no location header', done => {
+            bucketUtil = new BucketUtility('default', sigCfg);
+            s3 = bucketUtil.s3;
+            process.stdout.write('Creating bucket\n');
+            return s3.createBucket({ Bucket: bucket,
+                CreateBucketConfiguration: {
+                    LocationConstraint: 'mem',
+                },
+            }, err => {
+                if (err) {
+                    process.stdout.write(`Error creating bucket: ${err}\n`);
+                    return err;
+                }
+                process.stdout.write('Putting object\n');
+                return s3.putObject(params, err => {
+                    assert.equal(err, null, 'Expected success, ' +
+                        `got error ${JSON.stringify(err)}`);
+                    s3.getObject({ Bucket: bucket, Key: key }, err => {
+                        assert.strictEqual(err, null, 'Expected success, ' +
+                        `got error ${JSON.stringify(err)}`);
+                        done();
+                    });
+                });
+            });
+        });
+    });
+});
+
+describe('MultipleBackend put based on request endpoint', () => {
+    withV4(sigCfg => {
+        after(() => {
+            process.stdout.write('Deleting bucket\n');
+            return bucketUtil.deleteOne(bucket)
+            .catch(err => {
+                process.stdout.write(`Error in after: ${err}\n`);
+                throw err;
+            });
+        });
+
+        it('should create bucket in corresponding backend', done => {
+            bucketUtil = new BucketUtility('default', sigCfg);
+            s3 = bucketUtil.s3;
+            process.stdout.write('Creating bucket');
+            const request = s3.createBucket({ Bucket: bucket });
+            request.on('build', () => {
+                request.httpRequest.body = '';
+            });
+            request.send(err => {
+                if (err) {
+                    process.stdout.write(`Error creating bucket: ${err}\n`);
+                    throw err;
+                }
+                const host = request.service.endpoint.hostname;
+                const endpoint = config.restEndpoints[host];
+                s3.getBucketLocation({ Bucket: bucket }, (err, data) => {
+                    assert.strictEqual(err, null, 'Expected succes, ' +
+                        `got error ${JSON.stringify(err)}`);
+                    assert.strictEqual(data.LocationConstraint, endpoint);
+                    done();
+                });
+            });
+        });
```

Comment:
we should also test mpu.  instead of doing "manual mpu's" like in some of our tests, you could use the node aws sdk upload call with a proper size object.